don't add -lssl etc if not needed (because we use system-postgresql, system-mysql)

diff --git a/connectivity/Library_postgresql-sdbc-impl.mk b/connectivity/Library_postgresql-sdbc-impl.mk
index c372f3f0c6f6..0343adb24bd8 100644
--- a/connectivity/Library_postgresql-sdbc-impl.mk
+++ b/connectivity/Library_postgresql-sdbc-impl.mk
@@ -46,7 +46,6 @@ $(eval $(call gb_Library_use_externals,postgresql-sdbc-impl,\
 	boost_headers \
 	postgresql \
 	$(if $(filter-out MSC,$(COM)), \
-		openssl \
 		$(if $(ENABLE_LDAP),openldap) \
 		nss3 \
 		plc4 \

diff --git a/connectivity/Library_mysqlc.mk b/connectivity/Library_mysqlc.mk
index f765fb1e305f..0999b46d3eec 100644
--- a/connectivity/Library_mysqlc.mk
+++ b/connectivity/Library_mysqlc.mk
@@ -12,7 +12,6 @@ $(eval $(call gb_Library_Library,mysqlc))
 $(eval $(call gb_Library_use_externals,mysqlc,\
 	boost_headers \
 	mariadb-connector-c \
-	openssl \
 ))
 
 $(eval $(call gb_Library_set_include,mysqlc,\

same reason as with postgresql and mysql?
 
diff --git a/ucb/CppunitTest_ucb_webdav_local_neon.mk b/ucb/CppunitTest_ucb_webdav_local_neon.mk
index 2e920f4ca03f..a4ea2358d2fe 100644
--- a/ucb/CppunitTest_ucb_webdav_local_neon.mk
+++ b/ucb/CppunitTest_ucb_webdav_local_neon.mk
@@ -32,7 +32,6 @@ $(eval $(call gb_CppunitTest_use_externals,ucb_webdav_local_neon,\
 	boost_headers \
 	libxml2 \
 	neon \
-	openssl \
 ))
 
 $(eval $(call gb_CppunitTest_use_custom_headers,ucb_webdav_local_neon,\
diff --git a/ucb/Library_ucpdav1.mk b/ucb/Library_ucpdav1.mk
index b9e1d1037897..a7bba12787fd 100644
--- a/ucb/Library_ucpdav1.mk
+++ b/ucb/Library_ucpdav1.mk
@@ -35,7 +35,6 @@ $(eval $(call gb_Library_use_externals,ucpdav1,\
 	boost_headers \
 	libxml2 \
 	neon \
-	openssl \
 ))
 
 $(eval $(call gb_Library_use_custom_headers,ucpdav1,\
@@ -72,7 +71,6 @@ else # WITH_WEBDAV == serf
 $(eval $(call gb_Library_use_externals,ucpdav1,\
 	boost_headers \
 	apr \
-	openssl \
 	serf \
 	zlib \
 ))
diff --git a/ucb/CppunitTest_ucb_webdav_neon_opts.mk b/ucb/CppunitTest_ucb_webdav_neon_opts.mk
index df670ed7afdd..020d9f9b6dd4 100644
--- a/ucb/CppunitTest_ucb_webdav_neon_opts.mk
+++ b/ucb/CppunitTest_ucb_webdav_neon_opts.mk
@@ -35,7 +35,6 @@ $(eval $(call gb_CppunitTest_use_externals,ucb_webdav_neon_opts,\
 	boost_headers \
 	libxml2 \
 	neon \
-	openssl \
 ))
 
 $(eval $(call gb_CppunitTest_use_custom_headers,ucb_webdav_neon_opts,\
diff --git a/ucb/CppunitTest_ucb_webdav_propfindcache.mk b/ucb/CppunitTest_ucb_webdav_propfindcache.mk
index 8d4a12d32ad4..8504a0200ae7 100644
--- a/ucb/CppunitTest_ucb_webdav_propfindcache.mk
+++ b/ucb/CppunitTest_ucb_webdav_propfindcache.mk
@@ -35,7 +35,6 @@ $(eval $(call gb_CppunitTest_use_externals,ucb_webdav_propfindcache,\
 	boost_headers \
 	libxml2 \
 	neon \
-	openssl \
 ))
 
 $(eval $(call gb_CppunitTest_use_custom_headers,ucb_webdav_propfindcache,\
diff --git a/ucb/CppunitTest_ucb_webdav_res_access.mk b/ucb/CppunitTest_ucb_webdav_res_access.mk
index fa0669987bd2..ad25d29f945b 100644
--- a/ucb/CppunitTest_ucb_webdav_res_access.mk
+++ b/ucb/CppunitTest_ucb_webdav_res_access.mk
@@ -35,7 +35,6 @@ $(eval $(call gb_CppunitTest_use_externals,ucb_webdav_res_access,\
 	boost_headers \
 	libxml2 \
 	neon \
-	openssl \
 ))
 
 $(eval $(call gb_CppunitTest_use_custom_headers,ucb_webdav_res_access,\
... and don't uselessly check for openssl in configure

diff --git a/config_host.mk.in b/config_host.mk.in
index 42f4511a62e8..0d7877fa9e9d 100644
--- a/config_host.mk.in
+++ b/config_host.mk.in
@@ -460,8 +460,6 @@ export OFFICEOTRON=@OFFICEOTRON@
 export BFFVALIDATOR=@BFFVALIDATOR@
 export OOO_JUNIT_JAR=@OOO_JUNIT_JAR@
 export OOO_VENDOR=@OOO_VENDOR@
-export OPENSSL_CFLAGS=$(gb_SPACE)@OPENSSL_CFLAGS@
-export OPENSSL_LIBS=$(gb_SPACE)@OPENSSL_LIBS@
 export OPENSYMBOL_TTF=@OPENSYMBOL_TTF@
 export ORCUS_CFLAGS=$(gb_SPACE)@ORCUS_CFLAGS@
 export ORCUS_LIBS=$(gb_SPACE)@ORCUS_LIBS@
@@ -608,7 +606,6 @@ export SYSTEM_NSS=@SYSTEM_NSS@
 export SYSTEM_ODBC_HEADERS=@SYSTEM_ODBC_HEADERS@
 export SYSTEM_ODFGEN=@SYSTEM_ODFGEN@
 export SYSTEM_OPENLDAP=@SYSTEM_OPENLDAP@
-export SYSTEM_OPENSSL=@SYSTEM_OPENSSL@
 export SYSTEM_PAGEMAKER=@SYSTEM_PAGEMAKER@
 export SYSTEM_POPPLER=@SYSTEM_POPPLER@
 export SYSTEM_POSTGRESQL=@SYSTEM_POSTGRESQL@
diff --git a/configure.ac b/configure.ac
index e333538a2601..3ab8f58cd507 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10654,64 +10654,9 @@ fi
 dnl ===================================================================
 dnl Check for system openssl
 dnl ===================================================================
-DISABLE_OPENSSL=
-AC_MSG_CHECKING([whether to disable OpenSSL usage])
-if test "$enable_openssl" = "yes"; then
-    AC_MSG_RESULT([no])
-    if test "$_os" = Darwin ; then
-        # OpenSSL is deprecated when building for 10.7 or later.
-        #
-        # http://stackoverflow.com/questions/7406946/why-is-apple-deprecating-openssl-in-macos-10-7-lion
-        # http://stackoverflow.com/questions/7475914/libcrypto-deprecated-on-mac-os-x-10-7-lion
-
-        with_system_openssl=no
-        libo_CHECK_SYSTEM_MODULE([openssl],[OPENSSL],[openssl])
-    elif test "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o "$_os" = "OpenBSD" -o "$_os" = "DragonFly" \
-            && test "$with_system_openssl" != "no"; then
-        with_system_openssl=yes
-        SYSTEM_OPENSSL=TRUE
-        OPENSSL_CFLAGS=
-        OPENSSL_LIBS="-lssl -lcrypto"
-    else
-        libo_CHECK_SYSTEM_MODULE([openssl],[OPENSSL],[openssl])
-    fi
-    if test "$with_system_openssl" = "yes"; then
-        AC_MSG_CHECKING([whether openssl supports SHA512])
-        AC_LANG_PUSH([C])
-        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <openssl/sha.h>]],[[
-            SHA512_CTX context;
-]])],[AC_MSG_RESULT([yes])],[AC_MSG_ERROR([no, openssl too old. Need >= 0.9.8.])])
-        AC_LANG_POP(C)
-    fi
-else
-    AC_MSG_RESULT([yes])
-    DISABLE_OPENSSL=TRUE
-
-    # warn that although OpenSSL is disabled, system libraries may depend on it
-    AC_MSG_WARN([OpenSSL has been disabled. No code compiled here will make use of it but system libraries may create indirect dependencies])
-    add_warning "OpenSSL has been disabled. No code compiled here will make use of it but system libraries may create indirect dependencies"
-fi
-
+DISABLE_OPENSSL=TRUE
 AC_SUBST([DISABLE_OPENSSL])
 
-if test "$enable_cipher_openssl_backend" = yes && test "$DISABLE_OPENSSL" = TRUE; then
-    if test "$libo_fuzzed_enable_cipher_openssl_backend" = yes; then
-        AC_MSG_NOTICE([Resetting --enable-cipher-openssl-backend=no])
-        enable_cipher_openssl_backend=no
-    else
-        AC_MSG_ERROR([--enable-cipher-openssl-backend needs OpenSSL, but --disable-openssl was given.])
-    fi
-fi
-AC_MSG_CHECKING([whether to enable the OpenSSL backend for rtl/cipher.h])
-ENABLE_CIPHER_OPENSSL_BACKEND=
-if test "$enable_cipher_openssl_backend" = yes; then
-    AC_MSG_RESULT([yes])
-    ENABLE_CIPHER_OPENSSL_BACKEND=TRUE
-else
-    AC_MSG_RESULT([no])
-fi
-AC_SUBST([ENABLE_CIPHER_OPENSSL_BACKEND])
-
 dnl ===================================================================
 dnl Check for building gnutls
 dnl ===================================================================

.. and without these gbuild still wants it ...

diff --git a/external/Module_external.mk b/external/Module_external.mk
index 1c722a325c2d..3e26c2c3db7b 100644
--- a/external/Module_external.mk
+++ b/external/Module_external.mk
@@ -81,7 +81,6 @@ $(eval $(call gb_Module_add_moduledirs,external,\
 	$(call gb_Helper_optional,NSS,nss) \
 	$(call gb_Helper_optional,ODFGEN,libodfgen) \
 	$(call gb_Helper_optional,OPENLDAP,openldap) \
-	$(call gb_Helper_optional,OPENSSL,openssl) \
 	$(call gb_Helper_optional,ORCUS,liborcus) \
 	$(call gb_Helper_optional,OWNCLOUD_ANDROID_LIB,owncloud-android-lib) \
 	$(call gb_Helper_optional,PAGEMAKER,libpagemaker) \
diff --git a/oox/CppunitTest_oox_tokenmap.mk b/oox/CppunitTest_oox_tokenmap.mk
index 8019da7a2301..0c6ab952a813 100644
--- a/oox/CppunitTest_oox_tokenmap.mk
+++ b/oox/CppunitTest_oox_tokenmap.mk
@@ -20,19 +20,10 @@ $(eval $(call gb_CppunitTest_use_sdk_api,oox_tokenmap))
 
 $(eval $(call gb_CppunitTest_use_library_objects,oox_tokenmap,oox))
 
-ifeq ($(TLS),OPENSSL)
-$(eval $(call gb_CppunitTest_use_externals,oox_tokenmap,\
-	openssl \
-	openssl_headers \
-))
-else
-ifeq ($(TLS),NSS)
 $(eval $(call gb_CppunitTest_use_externals,oox_tokenmap,\
        plc4 \
        nss3 \
 ))
-endif
-endif
 
 $(eval $(call gb_CppunitTest_use_libraries,oox_tokenmap,\
     avmedia \
diff --git a/oox/Library_oox.mk b/oox/Library_oox.mk
index 287a4bb70003..fc1e3d4d3cd3 100644
--- a/oox/Library_oox.mk
+++ b/oox/Library_oox.mk
@@ -67,19 +67,10 @@ $(eval $(call gb_Library_use_externals,oox,\
 	boost_headers \
 ))
 
-ifeq ($(TLS),OPENSSL)
-$(eval $(call gb_Library_use_externals,oox,\
-	openssl \
-	openssl_headers \
-))
-else
-ifeq ($(TLS),NSS)
 $(eval $(call gb_Library_use_externals,oox,\
        plc4 \
        nss3 \
 ))
-endif
-endif
 
 $(eval $(call gb_Library_set_componentfile,oox,oox/util/oox))
 
diff --git a/RepositoryExternal.mk b/RepositoryExternal.mk
index ad3febbae41b..70a5b4959b64 100644
--- a/RepositoryExternal.mk
+++ b/RepositoryExternal.mk
@@ -1546,75 +1546,11 @@ endef
 
 endif # SYSTEM_HARFBUZZ
 
-ifeq ($(DISABLE_OPENSSL),TRUE)
 
 gb_ExternalProject__use_openssl:=
 gb_LinkTarget__use_openssl_headers:=
 gb_LinkTarget__use_openssl:=
 
-else # !DISABLE_OPENSSL
-
-ifneq ($(SYSTEM_OPENSSL),)
-
-gb_LinkTarget__use_openssl_headers:=
-gb_ExternalProject__use_openssl:=
-
-define gb_LinkTarget__use_openssl
-$(call gb_LinkTarget_set_include,$(1),\
-	$$(INCLUDE) \
-	$(OPENSSL_CFLAGS) \
-)
-$(call gb_LinkTarget_add_libs,$(1),$(OPENSSL_LIBS))
-
-endef
-
-else # !SYSTEM_OPENSSL
-
-$(eval $(call gb_Helper_register_packages_for_install,ooo, \
-	openssl \
-))
-
-define gb_ExternalProject__use_openssl
-$(call gb_ExternalProject_use_package,$(1),openssl)
-
-endef
-
-define gb_LinkTarget__use_openssl_headers
-$(call gb_LinkTarget_use_external_project,$(1),openssl,full)
-$(call gb_LinkTarget_set_include,$(1),\
-	-I$(call gb_UnpackedTarball_get_dir,openssl)/include \
-	$$(INCLUDE) \
-)
-
-endef
-
-define gb_LinkTarget__use_openssl
-$(call gb_LinkTarget_use_package,$(1),openssl)
-ifeq ($(OS),WNT)
-$(call gb_LinkTarget_add_libs,$(1),\
-	$(call gb_UnpackedTarball_get_dir,openssl)/libcrypto.lib \
-	$(call gb_UnpackedTarball_get_dir,openssl)/libssl.lib \
-)
-else
-$(call gb_LinkTarget_add_libs,$(1),\
-	$(call gb_UnpackedTarball_get_dir,openssl)/libssl.a \
-	$(call gb_UnpackedTarball_get_dir,openssl)/libcrypto.a \
-	$(if $(filter $(OS),LINUX),-pthread) \
-)
-$(call gb_LinkTarget_use_external_project,$(1),openssl)
-ifeq ($(OS),SOLARIS)
-$(call gb_LinkTarget_add_libs,$(1),\
-	-lnsl \
-	-lsocket \
-)
-endif
-endif
-
-endef
-
-endif # SYSTEM_OPENSSL
-endif # DISABLE_OPENSSL
-
 
 ifeq ($(DISABLE_OPENSSL),TRUE)
 
diff --git a/external/neon/Library_neon.mk b/external/neon/Library_neon.mk
index 4c034a3178fb..b22d5240e7b3 100644
--- a/external/neon/Library_neon.mk
+++ b/external/neon/Library_neon.mk
@@ -15,8 +15,6 @@ $(eval $(call gb_Library_use_externals,neon,\
 	gnutls \
 	libxml2 \
 	libgcrypt \
-	openssl \
-	openssl_headers \
 	zlib \
 ))
 

